recursio-client/Managers/ClientGhostManager.gd

```diff
@@ -0,0 +1,174 @@
+extends GhostManager
+
+var _enemy_ghost_scene = preload("res://Characters/Ghost.tscn")
+var _player_ghost_scene = preload("res://Characters/PlayerGhost.tscn")
+
+# holds only the player ghosts
+var _player_ghosts: Array = []
+# holds only the enemy ghosts
+var _enemy_ghosts: Array = []
+
+# OVERRIDE #
+func on_preparation_phase_started() -> void:
+	.on_preparation_phase_started()
+	_non_vfx_spawn_ghosts()
+	_toggle_ghost_animation(false)
+	_toggle_visbility_lights(false)
+	# Display paths of my ghosts
+	_update_ghost_paths()
+
+# OVERRIDE #
+func on_countdown_phase_started() -> void:
+	.on_countdown_phase_started()
+	# Delete ghost path visualization
+	for ghost in _player_ghosts:
+		ghost.delete_path()
+	_toggle_ghost_animation(true)
+	_visual_kill_ghosts()
+	var countdown_phase_seconds = Constants.get_value("gameplay","countdown_phase_seconds")
+	var spawn_time = Constants.get_value("gameplay","spawn_time")
+	_visual_delay_spawn_ghosts(countdown_phase_seconds-spawn_time)
+
+# OVERRIDE #
+func on_game_phase_started() -> void:
+	.on_game_phase_started()
+	_toggle_visbility_lights(true)
+	_toggle_ghost_animation(true)
+
+# OVERRIDE #
+func on_game_phase_stopped() -> void:
+	.on_game_phase_stopped()
+
+# OVERRIDE #
```

Comment:
Since there are multiple complete overrides like this that don't actually use the base function, I feel like it might be more fitting to have an abstract GhostManager interface which is implemented separately on server and client... but no big need to change it imo, just an idea.

Reply:
I agree, but as the ghost handling is encapsulated now this should be a quick fix at a future date and I wanted to get this branch rather soonish, so the merge with the tutorial branch won't be too painful.

Reply:
Yeah I agree, no need to change it immediately.